src/transformStateWithClones.js

```
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedState = Object.assign({}, state);
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(clonedState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(clonedState, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(clonedState);
+        break;
+
+      default:
+        return 'Error';
```

Issue: When an unknown action type is encountered, the function returns the string 'Error'. According to the checklist, you should throw an error instead of returning a string. Please update this to `throw new Error('Unknown action type')` or similar.